lightly_studio/src/lightly_studio/core/dataset.py

```diff
@@ -288,6 +303,36 @@ def add_samples_from_path(
             image_paths=image_paths,
         )
 
+        if tag_depth == 1 and created_sample_ids:
+            print(f"Adding directory tags to {len(created_sample_ids)} new samples.")
+            parent_dir_to_sample_ids = defaultdict(list)
+
+            # Fetch newly created samples to get their file paths
+            newly_created_samples = (
+                self.query()
+                .match(SampleField.id.is_in(created_sample_ids))
+                .to_list()
+            )
+
+            for sample in newly_created_samples:
+                parent_dir = Path(sample.file_path_abs).parent.name.strip() 
```

Comment:
This will not get the depth=1, but the depth=last directory.

E.g. if `path = "/my/path"` and `sample.file_path_abs="/my/path/dir_depth_1/dir_depth_2/img.jpg`, then `parent_dir="dir_depth_2"`.
The solution is to rather use a logic similar to `Path(sample.file_path_abs).relative_to(path).split("/")[0]`.

However, you need to handle the case of an image being in the path directly, e.g. `path = "/my/path"` and `sample.file_path_abs="/my/path/img.jpg` should not lead to an error.